src/inmanta/parser/plyInmantaLex.py

```diff
@@ -132,7 +134,24 @@ def t_JCOMMENT(t: lex.LexToken) -> None:  # noqa: N802
 
 def t_MLS(t: lex.LexToken) -> lex.LexToken:
     r'"{3,5}([\s\S]*?)"{3,5}'
-    value = bytes(t.value[3:-3], "utf-8").decode("unicode_escape")
+
+    #   Check for the presence of an invalid escape sequence (ex: "\.")
+    #   Python < 3.12 raises a DeprecationWarning when encountering an invalid escape sequence
+    #   Python 3.12 will raise a SyntaxWarning
+    #   Future versions will eventually raise a SyntaxError
+    #   see ( https://docs.python.org/3.12/whatsnew/3.12.html#other-language-changes )
+    try:
+        with warnings.catch_warnings():
+            warnings.filterwarnings("error", message="invalid escape sequence")
```

Comment:
Let's make sure we only raise what we catch. If we still have this code in 3.12 we wouldn't want it to start raising a `SyntaxWarning` as an exception.
```suggestion
            warnings.filterwarnings("error", category=DeprecationWarning, message="invalid escape sequence")
```

Reply:
Makes sense